code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,648 @@
+		for(var/obj/O in range(5, owner))
+			O.cult_conceal()
+		revealing = TRUE
+		name = "Reveal Runes"
+		button_icon_state = "revealing"
+	else
+		owner.visible_message("<span class='warning'>A flash of light shines from [owner]'s hand!</span>", \
+			 "<span class='cultitalic'>You invoke the counterspell, revealing nearby runes and cult structures.</span>")
+		charges--
+		owner.whisper(invocation)
+		SEND_SOUND(owner, sound('sound/misc/enter_blood.ogg', 0, 1, 25))
+		for(var/obj/O in range(5, owner))
+			O.cult_reveal()
+		revealing = FALSE
+		name = "Conceal Runes"
+		button_icon_state = "veiling"
+	if(charges <= 0)
+		qdel(src)
+	desc = base_desc
+	desc += "<br><b><u>Has [charges] use\s remaining</u></b>."
+	UpdateButtonIcon()
+
+
+// The "magic hand" items
+/obj/item/melee/blood_magic
+	name = "\improper magical aura"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon = 'icons/obj/items.dmi'
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	flags = ABSTRACT | DROPDEL
+
+	w_class = WEIGHT_CLASS_HUGE
+	throwforce = 0
+	throw_range = 0
+	throw_speed = 0
+	var/invocation
+	var/uses = 1
+	var/health_cost = 0 //The amount of health taken from the user when invoking the spell
+	var/datum/action/innate/cult/blood_spell/source
+
+/obj/item/melee/blood_magic/New(loc, spell)
+	source = spell
+	uses = source.charges
+	health_cost = source.health_cost
+	..()
+
+/obj/item/melee/blood_magic/Destroy()
+	if(!QDELETED(source))
+		if(uses <= 0)
+			source.hand_magic = null
+			qdel(source)
+			source = null
+		else
+			source.hand_magic = null
+			source.charges = uses
+			source.desc = source.base_desc
+			source.desc += "<br><b><u>Has [uses] use\s remaining</u></b>."
+			source.UpdateButtonIcon()
+	..()
+
+/obj/item/melee/blood_magic/attack_self(mob/living/user)
+	afterattack(user, user, TRUE)
+
+/obj/item/melee/blood_magic/attack(mob/living/M, mob/living/carbon/user)
+	if(!iscarbon(user) || !iscultist(user))
+		uses = 0
+		qdel(src)
+		return
+	add_attack_logs(user, M, "used a cult spell ([src]) on")
+	M.lastattacker = user.real_name
+
+/obj/item/melee/blood_magic/afterattack(atom/target, mob/living/carbon/user, proximity)
+	. = ..()
+	if(invocation)
+		user.whisper(invocation)
+	if(health_cost && iscarbon(user))
+		var/mob/living/carbon/C = user
+		C.apply_damage(health_cost, BRUTE, pick("l_arm", "l_hand", "r_arm", "r_hand"))
+	if(uses <= 0)
+		qdel(src)
+	else if(source)
+		source.desc = source.base_desc
+		source.desc += "<br><b><u>Has [uses] use\s remaining</u></b>."
+		source.UpdateButtonIcon()
+
+//The spell effects
+
+//stun
+/obj/item/melee/blood_magic/stun
+	name = "Stunning Aura"
+	desc = "Will stun and mute a victim on contact."
+	color = RUNE_COLOR_RED
+	invocation = "Fuu ma'jin!"
+
+/obj/item/melee/blood_magic/stun/afterattack(atom/target, mob/living/carbon/user, proximity)
```

Comment:
This will currently bypass shields due to how shields work. afterattack is called even when the attack is blocked. Not sure if this is intended behaviour yet but it is currently how it works

Reply:
I am going to leave this as is for now as it is a spell. If updated cult needs nerfing, this is one of the nerfs that will be done first